deliver/lib/deliver/upload_metadata.rb

```diff
@@ -226,30 +226,47 @@ def verify_available_languages!(options)
       # Collect all languages we need
       # We only care about languages from user provided values
       # as the other languages are on iTC already anyway
-      v = options[:app].edit_version
-      UI.user_error!("Could not find a version to edit for app '#{options[:app].name}', the app metadata is read-only currently") unless v
+      v = options[:app].details
+      UI.user_error!("Could not find details for app '#{options[:app].name}'") unless v
+
+      # find languages currently enabled on ITC (enabled_languages)
+      enabled_languages = {}  # map of locale code to language data hash
+      primary_data = nil      # the primary language data, e.g. english name/subtitle
+      v.languages.each do |language_data|
+        lang = language_data["localeCode"]
+        enabled_languages[lang] = language_data
+        primary_data = language_data unless language_data["canDeleteLocale"]
+      end
 
-      enabled_languages = options[:languages] || []
+      UI.user_error!("could not find primary language data for app '#{options[:app].name}'") unless primary_data
+
+      # find list of languages we have data for (local_languages)
+      local_languages = options[:languages] || []
```

Comment:
what does `local` refer to here?

Reply:
here i was using `local` to refer to the list of languages defined under .fastlane/metadata